packages/pigeon/tool/shared/generation.dart

```diff
@@ -79,6 +79,7 @@ Future<int> generatePigeons({required String baseDir}) async {
           ? null
           : '$outputBase/android/src/main/kotlin/com/example/test_plugin/$pascalCaseName.gen.kt',
       kotlinPackage: 'com.example.test_plugin',
+      kotlinErrorClassName: '${pascalCaseName}Error',
```

Comment:
I'd like to have end-to-end testing of the default behavior too; let's make the value here `input == 'core_tests' ? null : '${pascalCaseName}Error'`.